File: KevinRene/src/ql/ast/expression/arithmetic/Subtract.java
package ql.ast.expression.arithmetic;

import ql.ast.Expression;
import ql.ast.QLType;
import ql.ast.expression.Binary;
import ql.ast.type.QLNumeric;
import ql.ast.visitor.ExpressionVisitor;

public class Subtract extends Binary {
	public Subtract(Expression left, Expression right) {
		super(left, right, "-");
	}
	
	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {		
		return visitor.visit(this);
	}
	
	@Override
	public QLType getType() {
		return new QLNumeric();
	}
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression/Binary/Subtract.cs
﻿using AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes.Expression.Binary
{
    public class Subtract : Binary, IExpression
    {
        public Subtract(IExpression left, IExpression right, PositionInText position)
            : base(left, right, position)
        { }

        public override T Accept<T>(ASTVisitors.IVisitor<T> visitor)
        {
           return visitor.Visit(this);
        }
        public override string ToString()
        {
            return "-";
        }
    }
}